python/tank_vendor/shotgun_authentication/session_cache.py

```diff
@@ -67,81 +66,242 @@ def _get_local_site_cache_location(base_url):
         root = os.path.join(os.environ["APPDATA"], "Shotgun")
     elif sys.platform.startswith("linux"):
         root = os.path.expanduser("~/.shotgun")
+    return root
+
+
+def _get_local_site_cache_location(base_url):
+    """
+    Returns the location of the site cache root based on a site.
+
+    :param base_url: Site we need to compute the root path for.
+
+    :returns: An absolute path to the site cache root.
+    """
+    return os.path.join(
+        _get_cache_location(),
+        # get site only; https://www.foo.com:8080 -> www.foo.com
+        urlparse.urlparse(base_url)[1].split(":")[0]
+    )
 
-    # get site only; https://www.foo.com:8080 -> www.foo.com
-    base_url = urlparse.urlparse(base_url)[1].split(":")[0]
 
-    return os.path.join(root, base_url)
+def _get_authentication_cache_location(base_url):
+    """
+    Returns the location for the authentication cache folder for a given site.
+
+    :param base_url: Site we need to compute the authentication cache path for.
+
+    :returns: An absolute path to the authentication cache root.
+    """
+    return os.path.join(
+        _get_local_site_cache_location(base_url),
+        "authentication"
+    )
 
 
-def _get_session_data_location(base_url):
+def _get_users_file_location(base_url):
     """
-    Returns the location of the session file on disk for a specific site.
+    Returns the location of the users file on disk for a specific site.
 
     :param base_url: The site we want the login information for.
 
     :returns: Path to the login information.
     """
     return os.path.join(
-        _get_local_site_cache_location(base_url),
+        _get_authentication_cache_location(base_url),
+        "users.yml"
+    )
+
+
+def _get_current_user_file_location(base_url):
+    """
+    Returns the location of the users file on disk for a specific site.
+
+    :param base_url: The site we want the login information for.
+
+    :returns: Path to the login information.
+    """
+    return os.path.join(
+        _get_authentication_cache_location(base_url),
+        "current_user.yml"
```

Comment:
Setting the current user and caching the current user is done through two different code paths (The SessionUser.__init__ method to cache the user vs the DefaultsManager.set_login) by two unrelated sections of the code. I figured since I would end up opening and write the file twice I might as well have two files. It makes QA-ing easier too. But I agree, if we want this to grow more organically, maybe it's better if we put everything in the same file under a more generic term so that it can easily be updated in the future without having to worry about a bunch of files.

One caveat though: I really would prefer having a subfolder for authentication. It is already getting crowded in that folder since some OSes don't have a folder specific for caches and logs. Having a subfolder gives us a playground that wouldn't contribute to the mess if we ever start having multiple files to handle authentication.